@here/harp-mapview/lib/text/TextStyleCache.ts

```diff
@@ -84,50 +84,45 @@ export class TextStyleCache {
         layoutParams: defaultTextLayoutStyle.params
     };
 
-    constructor(
-        private m_textStyleDefinitions?: TextStyleDefinition[],
-        private readonly m_defaultTextStyleDefinition?: TextStyleDefinition
-    ) {}
-
-    initializeDefaultTextElementStyle() {
-        if (this.m_textStyleDefinitions === undefined) {
-            this.m_textStyleDefinitions = [];
-        }
-        const styles = this.m_textStyleDefinitions;
-        this.m_defaultStyle.fontCatalog = undefined;
+    constructor() {
+        this.updateDefaultTextStyle();
+    }
 
-        const themedDefaultStyle = styles.find(style => style.name === DEFAULT_STYLE_NAME);
-        if (themedDefaultStyle !== undefined) {
-            this.m_defaultStyle = this.createTextElementStyle(
-                themedDefaultStyle,
-                DEFAULT_STYLE_NAME
-            );
-        } else if (this.m_defaultTextStyleDefinition !== undefined) {
-            this.m_defaultStyle = this.createTextElementStyle(
-                this.m_defaultTextStyleDefinition,
-                DEFAULT_STYLE_NAME
-            );
-        } else if (styles.length > 0) {
-            this.m_defaultStyle = this.createTextElementStyle(styles[0], DEFAULT_STYLE_NAME);
+    updateTextStyles(
+        textStyleDefinitions?: TextStyleDefinition[],
+        defaultTextStyleDefinition?: TextStyleDefinition
+    ) {
+        if (!textStyleDefinitions || textStyleDefinitions.length === 0) {
+            this.m_textStyles.clear();
+        } else {
+            // Update Styles
+            textStyleDefinitions?.forEach(element => {
+                this.m_textStyles.set(
+                    element.name!,
+                    this.createTextElementStyle(element, element.name!)
+                );
+            });
+
+            // Remove obsolete
+            for (const [name] of this.m_textStyles) {
+                if (
+                    textStyleDefinitions.findIndex(definition => {
+                        return definition.name === name;
+                    }) === -1
+                ) {
+                    this.m_textStyles.delete(name);
+                }
+            }
```

Comment:
```suggestion
            this.m_textStyles.clear();
            textStyleDefinitions.forEach(element => {
                this.m_textStyles.set(
                    element.name!,
                    this.createTextElementStyle(element, element.name!)
                );
            });
```

Reply:
- textStyleDefinitions is always defined in the else branch of the condition statement.
- Why not just clearing the text styles map before adding the new ones. Since we're overwriting with the new text styles, it should be the same.

Reply:
done